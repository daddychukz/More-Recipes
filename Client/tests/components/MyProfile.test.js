import React from 'react';
import { mount, shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import sinon from 'sinon';
import render from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import ConnectedUserProfile,
{ PureProfile } from '../../src/components/pages/MyProfile';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const initialState = {
  user: {},
  recipeOperations: {
    pagination: {},
    recipes: []
  },
  review: [],
  favorite: [],
  popularRecipes: [],
  singleRecipe: {},
  allRecipe: {
    pagination: {},
    recipes: []
  }
};
const store = mockStore(initialState);

const props = {
  getUserProfile: sinon.spy(() => new Promise((cb) => {
    cb();
  })),
  updateUserProfile: sinon.spy(() => new Promise((cb) => {
    cb();
  })),
  logout: sinon.spy(() => new Promise((cb) => {
    cb();
  })),
  profile: {
    userId: '',
    email: '',
    username: '',
    address: '',
    about: '',
    hobbies: '',
    phone: '',
    publicUrl: '',
    imageUrl: '',
    fullname: ''
  },
  resetPassword: sinon.spy(() => new Promise((cb) => {
    cb();
  }))
};

const mountedWrapper = mount(
  <BrowserRouter>
    <Provider store={store}>
      <ConnectedUserProfile {...props} />
    </Provider>
  </BrowserRouter>
);

const shallowWrapper = shallow(<PureProfile {...props} />);
const state = {
  FullName: props.profile.fullname,
  Phone: props.profile.phone,
  Address: props.profile.address,
  About: props.profile.about,
  Hobbies: props.profile.hobbies,
  PublicId: props.profile.publicUrl,
  Email: props.profile.email,
  oldPassword: '',
  newPassword: '',
  confirmPassword: '',
  isLoading: false
};
const event = {
  preventDefault: jest.fn(),
  target: {
    Review: '',
    Category: ''
  }
};

describe('MyFavorite test suite', () => {
  it('renders MyFavorite component without crashing', () => {
    expect(mountedWrapper).toBeDefined();
    expect(mountedWrapper.find('MyProfile').length).toBe(1);
  });

  it('should match component snapshot', () => {
    const tree = render.create(
      <BrowserRouter >
        <Provider store={store}>
          <PureProfile {...props} />
        </Provider>
      </BrowserRouter>);
    expect(tree).toMatchSnapshot();
  });

  it('calls logout event', () => {
    sinon.spy(shallowWrapper.instance(), 'logout');
    shallowWrapper.instance().logout();
    expect(shallowWrapper.instance().logout.calledOnce).toEqual(true);
  });

  it('calls onSubmit event', () => {
    sinon.spy(shallowWrapper.instance(), 'onSubmit');
    shallowWrapper.setState(state);
    shallowWrapper.instance().onSubmit(event);
    expect(shallowWrapper.instance().onSubmit.calledOnce).toEqual(true);
  });

  it('calls onChange event', () => {
    sinon.spy(shallowWrapper.instance(), 'onChange');
    shallowWrapper.setState(state);
    shallowWrapper.instance().onChange(event);
    expect(shallowWrapper.instance().onChange.calledOnce).toEqual(true);
  });

  it('calls uploadWidget function when user adds image successfully', () => {
    sinon.spy(shallowWrapper.instance(), 'uploadWidget');
    shallowWrapper.instance().uploadWidget();
    expect(shallowWrapper.instance().uploadWidget.calledOnce).toEqual(true);
  });

  it('throws error user password fields doesnt match', () => {
    state.newPassword = '123';
    state.confirmPassword = '223';
    sinon.spy(shallowWrapper.instance(), 'changePassword');
    shallowWrapper.setState(state);
    shallowWrapper.instance().changePassword(event);
    expect(shallowWrapper.instance().changePassword.calledOnce).toEqual(true);
  });

  it('dispatches action to change user password', () => {
    state.newPassword = '123';
    state.confirmPassword = '123';
    shallowWrapper.setState(state);
    shallowWrapper.instance().changePassword(event);
    expect(shallowWrapper.instance().changePassword.called).toEqual(true);
  });

  it('calls change password modal', () => {
    sinon.spy(shallowWrapper.instance(), 'openModal');
    shallowWrapper.setState(state);
    shallowWrapper.instance().openModal();
    expect(shallowWrapper.instance().openModal.calledOnce).toEqual(true);
  });

  it('displays default user image', () => {
    props.profile.publicUrl = 'user-male_jvc8hn.jpg';
    sinon.spy(shallowWrapper.instance(), 'componentWillMount');
    shallowWrapper.setState(state);
    shallowWrapper.instance().componentWillMount();
    expect(shallowWrapper.instance().componentWillMount.calledOnce).toEqual(true);
  });
});
