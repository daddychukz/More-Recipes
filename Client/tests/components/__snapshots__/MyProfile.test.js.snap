// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyFavorite test suite should match component snapshot 1`] = `
<div>
  <div>
    <nav
      className="navbar navbar-expand-md bg-dark navbar-dark fixed-top"
    >
      <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="/recipe-box"
          onClick={[Function]}
        >
          <h3
            id="logo"
          >
            More-Recipes
          </h3>
        </a>
        <button
          className="navbar-toggler"
          data-target="#navbarCollapse"
          data-toggle="collapse"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarCollapse"
        >
          <ul
            className="navbar-nav ml-auto"
          >
            <li
              className="nav-item"
            >
              <a
                className="nav-link
                  false"
                href="/recipe-box"
                onClick={[Function]}
              >
                Home
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/add-recipe"
                onClick={[Function]}
              >
                Add Recipe
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/my-recipe"
                onClick={[Function]}
              >
                My Recipes
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/my-favorite"
                onClick={[Function]}
              >
                Favorites
              </a>
            </li>
            <li
              className="nav-item"
            >
              <div
                className="btn-group open"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-user-circle fa-2x pull-right drop"
                  data-toggle="dropdown"
                />
                <ul
                  className="dropdown-menu"
                >
                  <li>
                    <a
                      className="dropdown-item"
                      href="/my-profile"
                      onClick={[Function]}
                    >
                      Profile
                    </a>
                  </li>
                  <li
                    className="divider"
                  />
                  <li>
                    <a
                      className="dropdown-item"
                      data-target="#profileModal"
                      data-toggle="modal"
                      href="#"
                      id="profile-edit"
                      onClick={[Function]}
                    >
                      Edit Profile
                    </a>
                  </li>
                  <li>
                    <a
                      className="dropdown-item logOut"
                      href="/"
                      onClick={[Function]}
                    >
                      Logout
                    </a>
                  </li>
                </ul>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>
  <header
    id="header"
  />
  <div
    className="container"
  >
    <nav
      className="breadcrumb"
    >
      <a
        className="breadcrumb-item"
        href="/recipe-box"
        onClick={[Function]}
      >
        Home
      </a>
      <span
        className="breadcrumb-item active"
      >
        My Profile
      </span>
    </nav>
  </div>
  <section>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-4 d-none d-lg-block"
        >
          <div
            className="list-group mb-3"
          >
            <li
              className="list-group-item active text-center"
            >
              <h5
                id="Fname"
              />
            </li>
            <a
              className="list-group-item"
              href="/my-recipe"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="fa fa-cutlery"
                id="my-recipe"
              />
                My Recipes
              <span
                className="badge badge-pill badge-info float-right"
              >
                0
              </span>
            </a>
            <a
              className="list-group-item"
              href="/my-favorite"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="fa fa-star"
                id="my-favorite"
              />
                My Favourites
              <span
                className="badge badge-pill badge-info float-right"
                id="favCount"
              >
                0
              </span>
            </a>
            <a
              className="list-group-item"
              href="/my-profile"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="fa fa-user"
                id="my-profile"
              />
                My Profile
            </a>
          </div>
          <ul
            className="list-group mb-3"
          >
            <li
              className="list-group-item active text-center"
            >
              <h5>
                Popular Recipes
              </h5>
            </li>
          </ul>
        </div>
        <div
          className="loader"
        />
      </div>
    </div>
  </section>
  <div
    className="modal fade text-dark"
    id="changePassword"
  >
    <div
      className="modal-dialog"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header bg-info"
        >
          <h5
            className="modal-title"
            id="contactModalTitle"
            style={
              Object {
                "color": "white",
              }
            }
          >
            Change Your Password
          </h5>
        </div>
        <div
          className="modal-body"
        >
          <form>
            <div
              className="form-group"
            >
              <input
                className="form-control form-control-lg"
                name="oldPassword"
                onChange={[Function]}
                placeholder="Old Password"
                required={true}
                type="password"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control form-control-lg"
                name="newPassword"
                onChange={[Function]}
                placeholder="New Password"
                required={true}
                type="password"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control form-control-lg"
                name="confirmPassword"
                onChange={[Function]}
                placeholder="Confirm Password"
                required={true}
                type="password"
                value=""
              />
            </div>
            <input
              className="btn btn-info btn-block"
              data-dismiss="modal"
              onClick={[Function]}
              type="button"
              value="Change Password"
            />
          </form>
        </div>
      </div>
    </div>
  </div>
  <div
    className="modal fade text-dark"
    id="profileModal"
  >
    <div
      className="modal-dialog"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header bg-primary"
        >
          <h5
            className="modal-title"
            style={
              Object {
                "color": "white",
              }
            }
          >
            Edit Profile
          </h5>
        </div>
        <div
          className="modal-body"
        >
          <form
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="name"
              >
                Fullname
              </label>
              <input
                className="form-control"
                name="FullName"
                onChange={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="name"
              >
                Phone Number
              </label>
              <input
                className="form-control"
                name="Phone"
                onChange={[Function]}
                required={true}
                type="number"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="email"
              >
                Hobbies
              </label>
              <input
                className="form-control"
                name="Hobbies"
                onChange={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="name"
              >
                About
              </label>
              <input
                className="form-control"
                name="About"
                onChange={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="messager"
              >
                Address
              </label>
              <textarea
                className="form-control"
                name="Address"
                onChange={[Function]}
                required={true}
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <div>
                <img
                  src=""
                />
              </div>
              <input
                className="form-control-file"
                onClick={[Function]}
                type="button"
                value="Upload Image"
              />
              <small
                className="form-text text-muted"
                id="fileHelp"
              >
                Upload profile picture
              </small>
            </div>
            <div
              className="modal-footer"
            >
              <button
                className="mt-2 btn btn-primary btn-block"
                type="submit"
              >
                Update
              </button>
              <button
                className="btn btn-primary btn-block"
                data-dismiss="modal"
                type="button"
              >
                Close
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
